### PR TITLE
Add a page for HTMLElement.inputMode

### LINKED ISSUE #22003: Missing page and short description for HTMLElement.inputMode
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement

### What specific section or headline is this issue about?

browser_compatibility

### What information was incorrect, unhelpful, or incomplete?

https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/inputMode is linked in the table, but not available as a page; neither is it shortly described on the page itself.

### What did you expect to see?

A description of the property.

### Do you have any supporting links, references, or citations?

_No response_

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/api/htmlelement`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/api/htmlelement/index.md
* Last commit: https://github.com/mdn/content/commit/1732201134d732af9bfd49ee9e10f7e2086cc783
* Document last modified: 2022-10-14T07:09:52.000Z

</details>

### PR DESCRIPTION
**Pull request by wbamberg**
Fixes https://github.com/mdn/content/issues/22003.

This PR adds a page for `HTMLElement.inputMode`. Note that the spec table isn't showing because BCD is currently missing `spec_url`, so I filed https://github.com/mdn/browser-compat-data/pull/18137 for that.

### CODE REVIEW

**Comment by hamishwillee** [comment]
@wbamberg https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement still needs to be updated with a link to this right?

File: files/en-us/web/api/htmlelement/inputmode/index.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+This attribute may have one of the following values:
+
+- `decimal`
+  - : Fractional numeric input keyboard containing the digits and decimal separator for the user's locale (typically <kbd>.</kbd> or <kbd>,</kbd>).
+- `email`
+  - : A virtual keyboard optimized for entering email addresses.
+    Typically includes the <kbd>@</kbd>character as well as other optimizations.
+- `none`
+  - : No virtual keyboard, for when the page implements its own keyboard input control.
```
Comment (hamishwillee) [inline files/en-us/web/api/htmlelement/inputmode/index.md:34]: Personal taste thing, but I like to separate purpose and reason.

```suggestion
  - : No virtual keyboard. This is used when the page implements its own keyboard input control.
```

File: files/en-us/web/api/htmlelement/inputmode/index.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+- `numeric`
+  - : Numeric input keyboard, but only requires the digits 0–9.
```
Comment (hamishwillee) [inline files/en-us/web/api/htmlelement/inputmode/index.md:36]:
```suggestion
  - : Numeric input keyboard that only requires the digits 0–9.
```

File: files/en-us/web/api/htmlelement/inputmode/index.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    Devices may or may not show a minus key.
+- `search`
+  - : A virtual keyboard optimized for search input.
+    For instance, the [return/submit key](https://html.spec.whatwg.org/dev/interaction.html#input-modalities:-the-enterkeyhint-attribute) may be labeled "Search".
+- `tel`
+  - : A telephone keypad input, including the digits 0–9, the asterisk (<kbd>\*</kbd>), and the pound (<kbd>#</kbd>) key.
```
Comment (hamishwillee) [inline files/en-us/web/api/htmlelement/inputmode/index.md:42]:
```suggestion
  - : A telephone keypad input that includes the digits 0–9, the asterisk (<kbd>\*</kbd>), and the pound (<kbd>#</kbd>) key.
```

File: files/en-us/web/api/htmlelement/inputmode/index.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+It provides a hint about the type of data that might be entered by the user while editing the element or its contents. This allows the browser to display an appropriate virtual keyboard.
+
+It is used primarily on {{HTMLElement("input")}} elements, but is usable on any element in {{HTMLAttrxRef("contenteditable")}} mode.
+
+## Value
+
+This attribute may have one of the following values:
+
+- `decimal`
+  - : Fractional numeric input keyboard containing the digits and decimal separator for the user's locale (typically <kbd>.</kbd> or <kbd>,</kbd>).
```
Comment (hamishwillee) [inline files/en-us/web/api/htmlelement/inputmode/index.md:29]:
```suggestion
  - : Fractional numeric input keyboard that contains the digits and decimal separator for the user's locale (typically <kbd>.</kbd> or <kbd>,</kbd>).
```

**Review by hamishwillee — COMMENTED** [review]
Looks good. I've put in some comments "to my taste".

PS, I like how you put a little more information about this in the intro that just "reflects the global element"

File: files/en-us/web/api/htmlelement/inputmode/index.md (hunk 3 above)
Comment (wbamberg) [reply files/en-us/web/api/htmlelement/inputmode/index.md:42]: I copied all these from the attribute page so am not attached to them :).

**Comment by wbamberg** [comment]
> @wbamberg https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement still needs to be updated with a link to this right?

Oops, good point, done. Also the bit for `hidden` needed an update.

**Review by hamishwillee — APPROVED** [review]
LGTM!

**Pull request merged**